File: 2338-minimum-consecutive-cards-to-pick-up/2338-minimum-consecutive-cards-to-pick-up.py
class Solution:
    def minimumCardPickup(self, cards: List[int]) -> int:
        dct = dict()
        min_val = len(cards) + 1
        
        for (index, val) in enumerate(cards):
            if val in dct:
                min_val = min(min_val, index - dct[val] + 1)
            dct[val] = index
        print(min_val)
        if min_val >= len(cards) + 1:
            return -1
        return min_val

        

File: 2338-minimum-consecutive-cards-to-pick-up/2338-minimum-consecutive-cards-to-pick-up.rs
use std::collections::HashMap;
impl Solution {
    pub fn minimum_card_pickup(cards: Vec<i32>) -> i32 {
        let mut val_index_map : HashMap<i32, usize> = HashMap::new();
        let mut consective_min_len = cards.len() + 1;
        for (index, &val) in cards.iter().enumerate(){
           if let Some(&prev_index) = val_index_map.get(&val){
             consective_min_len = consective_min_len.min(index - prev_index + 1) 
           }
           val_index_map.insert(val, index);
        }
        if consective_min_len < cards.len() + 1{
            consective_min_len as i32
        }else{
            -1
        }
    }
}

//